### PR TITLE
Add docs for configuring OpenTelemetry for self-hosted

### PR DESCRIPTION
**Pull request by isabeldepapel**
## Description
Adds instructions along with some example configurations for setting up OpenTelemetry in self-hosted instances.

## Checklist:

- [x] I have reviewed the [style guide](https://github.com/pulumi/pulumi-hugo/blob/master/STYLE-GUIDE.md).
- [ ] If blogging, I have reviewed the [blogging guide](https://github.com/pulumi/pulumi-hugo/blob/master/BLOGGING.md).
- [x] I have manually confirmed that all new links work.
- [ ] I added aliases (i.e., redirects) for all filename changes.
- [ ] If making css changes, I rebuilt the bundle.

### CODE REVIEW

**Review by lblackstone — APPROVED** [review]

File: themes/default/content/docs/pulumi-cloud/self-hosted/components/api.md [hunk 1]
```diff
@@ -288,3 +288,150 @@ To use [AWS DynamoDB](https://aws.amazon.com/dynamodb) to persist Audit Logs, sp
...
+The following environment variables are needed to configure OpenTelemetry in the service:
+
+| Variable name | Description |
+| --- | --- |
+| OTEL_EXPORTER_OTLP_ENDPOINT | (Required) Used to configure the OTLP exporter. The base URL to which all telemetry will be sent. If not set, the API service will use no-op metrics and traces. |
+| OTEL_EXPORTER_OTLP_PROTOCOL | (Optional) Used to configure the OTLP exporter. Valid values are `http` or `grpc`. Defaults to `grpc`. |
+| PULUMI_ENABLE_DEPRECATED_METRICS | (Optional) Whether to continue emitting API service metrics in a log-based format. Defaults to `true`. |
+| METRICS_WEBHOOK_SECRET | Required to successfully authenticate to the `/metrics` endpoint. The Authorization header shoud be set as follows: `Authorization: webhook-token <METRICS_WEBHOOK_SECRET>`. |
+
+OpenTelemetry is not yet available for the node backend or console.
```
Comment (cnunciato) [inline themes/default/content/docs/pulumi-cloud/self-hosted/components/api.md:305]: These are the same thing in terms of services, right? (Apologies if not -- it's been a while.) If so, should we just call this "the console service" (as that's what it seems to be called in the sibling docs)?

```suggestion
OpenTelemetry is not yet available for the [console service](/docs/pulumi-cloud/self-hosted/components/console/).
```

File: themes/default/content/docs/pulumi-cloud/self-hosted/components/api.md [hunk 2]
```diff
@@ -62,7 +62,7 @@ It's possible to split apart the HTTP Server from the Background Jobs functional
 
 The core infrastructure components to successfully run the API service are the database, object storage, and encryption services.
 Depending on your requirements, you can configure additional (optional) identity services as well as enhanced security
-between the API and the database.
+between the API and the database. The API also supports exporting OpenTelemetry metrics and traces via the OpenTelemetry collector.
```
Comment (cnunciato) [inline themes/default/content/docs/pulumi-cloud/self-hosted/components/api.md:65]:
```suggestion
between the API and the database. The API also supports [exporting OpenTelemetry metrics and traces](#opentelemetry) via the OpenTelemetry collector.
```

File: themes/default/content/docs/pulumi-cloud/self-hosted/components/api.md [hunk 3]
```diff
@@ -288,3 +288,150 @@ To use [AWS DynamoDB](https://aws.amazon.com/dynamodb) to persist Audit Logs, sp
...
+The token type is `webhook-token`, not `Bearer`.
+{{% /notes %}}
+
+```sh
+curl -s GET https://api.pulumi.com/metrics -H 'Authorization: webhook-token <METRICS_WEBHOOK_SECRET>
+```
+
+### Prometheus
+
+The API service provides 2 options to get metrics into Prometheus:
```
Comment (cnunciato) [inline themes/default/content/docs/pulumi-cloud/self-hosted/components/api.md:321]:
```suggestion
The API service provides two options to get metrics into Prometheus:
```

File: themes/default/content/docs/pulumi-cloud/self-hosted/components/api.md [hunk 4]
```diff
@@ -288,3 +288,150 @@ To use [AWS DynamoDB](https://aws.amazon.com/dynamodb) to persist Audit Logs, sp
...
+      receivers: [otlp]
+      processors: [memory_limiter, batch]
+      exporters: [prometheusremotewrite]
+
+  extensions: [sigv4auth]
+```
+
+#### Prometheus exporter
+
+This option require configuring the environment variable `METRICS_WEBHOOK_SECRET` to successfully authenticate to the [`/metrics` endpoint](#metrics-endpoint)
```
Comment (cnunciato) [inline themes/default/content/docs/pulumi-cloud/self-hosted/components/api.md:375]:
```suggestion
This option requires configuring the environment variable `METRICS_WEBHOOK_SECRET` to successfully authenticate to the [`/metrics` endpoint](#metrics-endpoint).
```

File: themes/default/content/docs/pulumi-cloud/self-hosted/components/api.md [hunk 5]
```diff
@@ -288,3 +288,150 @@ To use [AWS DynamoDB](https://aws.amazon.com/dynamodb) to persist Audit Logs, sp
...
+| OTEL_EXPORTER_OTLP_ENDPOINT | (Required) Used to configure the OTLP exporter. The base URL to which all telemetry will be sent. If not set, the API service will use no-op metrics and traces. |
+| OTEL_EXPORTER_OTLP_PROTOCOL | (Optional) Used to configure the OTLP exporter. Valid values are `http` or `grpc`. Defaults to `grpc`. |
+| PULUMI_ENABLE_DEPRECATED_METRICS | (Optional) Whether to continue emitting API service metrics in a log-based format. Defaults to `true`. |
+| METRICS_WEBHOOK_SECRET | Required to successfully authenticate to the `/metrics` endpoint. The Authorization header shoud be set as follows: `Authorization: webhook-token <METRICS_WEBHOOK_SECRET>`. |
+
+OpenTelemetry is not yet available for the node backend or console.
+
+### Metrics endpoint
+
+The API service exposes a metrics endpoint (`https://api.pulumi.com/metrics`) that is secured by a bearer token. This token is configured using the environment variable, `METRICS_WEBHOOK_SECRET`.
```
Comment (cnunciato) [inline themes/default/content/docs/pulumi-cloud/self-hosted/components/api.md:309]:
```suggestion
The API service exposes a metrics endpoint (`https://api.pulumi.com/metrics`) that is secured by a bearer token. This token is configured using the environment variable `METRICS_WEBHOOK_SECRET`.
```

**Review by cnunciato — APPROVED** [review]
LGTM! A few little suggestions for you to consider.

**Pull request merged**